### PR TITLE
explore integration of @ngrx/router-store

### PR DESCRIPTION
**Pull request by dhhyi**
## PR Type

 [ ] Bugfix  
 [ ] Feature  
 [ ] Code style update (formatting, local variables)  
 [x] Refactoring (no functional changes, no API changes)  
 [ ] Build-related changes  
 [ ] CI-related changes  
 [ ] Documentation content changes  
 [ ] Application / infrastructure changes  
 [ ] Other: 

## What Is the Current Behavior?

Issue Number: 

## Does this PR Introduce a Breaking Change?

 [x] Yes  
 [ ] No

## TODO

* [x] replace ngrx-router with router-store in codebase
* [x] remove SelectProduct
* [x] remove SelectCategory
* [x] race-condition loading configuration & language in SSR mode
* [x] remove SelectSearchTerm and refactor ish-search-box
* [x] remove other SelectX actions
  * not possible for SelectOrder*, SelectQuote/SelectQuoteRequest, SelectWishlist
* [x] repair and check SEO Meta Data
* [x] add note to migration guide

~remove references to ActivatedRoute where possible~ -> https://github.com/intershop/intershop-pwa/projects/4#card-35976231

### CODE REVIEW

File: src/app/extensions/seo/guards/meta.guard.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  description: 'seo.defaults.description',
+  robots: 'index, follow',
+  'og:type': 'website',
+};
+
+@Injectable({ providedIn: 'root' })
+export class MetaGuard implements CanActivate, CanActivateChild {
+  constructor(private store: Store<{}>) {}
+
+  canActivate(route: ActivatedRouteSnapshot, _: RouterStateSnapshot): boolean {
```
Comment (SGrueber) [inline src/app/extensions/seo/guards/meta.guard.ts:20]: is this testable?

File: src/app/core/store/configuration/configuration.selectors.ts [hunk 2]
```diff
@@ -77,3 +77,16 @@ export const getServerConfigParameter = <T>(path: string) =>
         .split('.')
         .reduce((obj, key) => (obj && obj[key] !== undefined ? obj[key] : undefined), serverConfig as unknown) as T
   );
+
+export const getAvailableLocales = createSelector(
```
[the remark this answers is not in the document]
Comment (dhhyi) [reply src/app/core/store/configuration/configuration.selectors.ts:81]: As this is now dependent on entries in environment.ts, I'll just add the check that the initial state is not empty. 👍

File: src/app/shell/header/search-box/search-box.component.html [hunk 3]
```diff
@@ -1,17 +1,16 @@
-<form #searchForm role="search" class="search clearfix" name="SearchBox_Header" (submit)="submitSearch()">
+<form role="search" class="search clearfix" name="SearchBox_Header" (submit)="submitSearch(searchInput.value)">
```
[the remark this answers is not in the document]
Comment (dhhyi) [reply src/app/shell/header/search-box/search-box.component.html:1]: hmm, yes. I think we should add bugs later for this. The current behavior replicates the previous one backed by cypress tests.

File: src/app/core/store/orders/orders.effects.ts [hunk 4]
```diff
@@ -152,9 +151,9 @@ export class OrdersEffects {
...
   @Effect()
-  routeListenerForSelectingOrder$ = this.actions$.pipe(
-    ofRoute(/^(account\/orders.*|checkout\/receipt)/),
-    mapToParam<string>('orderId'),
+  routeListenerForSelectingOrder$ = this.store.pipe(
+    ofUrl(/^\/(account\/orders.*|checkout\/receipt)/),
+    select(selectRouteParam('orderId')),
     withLatestFrom(this.store.pipe(select(getSelectedOrderId))),
     filter(([fromAction, selectedOrderId]) => fromAction && fromAction !== selectedOrderId),
     map(([orderId]) => new ordersActions.SelectOrder({ orderId }))
```
[the remark this answers is not in the document]
Comment (dhhyi) [reply src/app/core/store/orders/orders.effects.ts:159]: As discussed `orderId` is supplied both by url params and query params. So I'm afraid of touching this 😅 

Technical debt added: https://github.com/intershop/intershop-pwa/projects/4#card-36343182

File: src/app/extensions/seo/guards/meta.guard.ts (hunk 1 above)
Comment (dhhyi) [reply src/app/extensions/seo/guards/meta.guard.ts:20]: Yes, but then the jest test would have to replicate all the routing and meta definitions. I guess it is enough to have the cypress tests here.

**Pull request merged**